### ISSUE TITLE
Support "Or" condition in qraphql query and REST API GET method

### ISSUE THREAD

**Issue by kwinyyyc** [amplication/amplication#2287]
### Feature description

The current source code generated by Amplication supports only "And" condition query like below for graphql and REST Api

Graphql Query
```gql
query {
  users(where: { firstName: { equals: "Test" }, lastName:{equals: "Test"}}) {
    firstName,
    createdAt,
    roles,
    id,
    lastName,
    updatedAt,
    username
  }
}
```

REST API GET query example:
```http://localhost:3000/api/users?where[firstName][equals]=Test&where[lastName][equals]=Test```

Both these requests regarded as getting user with firstName = "Test" And "lastName = "Test"

### Use case

For examples, I have an "Email" entity
{
   "from": "string",
   "to": "string"
}

As a user I would like to search for an email that sent to "abc company" or sent from "abc company"

**Comment by mshidlov** [comment]
Hey @kwinyyyc thank you for suggesting this enhancement, are you currently using Amplication and missing this feature?

**Comment by kwinyyyc** [comment]
@mshidlov Yes, but I find out it is a rare case to use "Or" condition so I have created a custom API to go around that. Can see if anyone else also need this feature.

**Comment by yuval-hazaz** [comment]
This issue will have a big impact on many areas and should not be approached until we have a design for it. 
I suggest keeping it open for future reference, but not approaching it before we have an approved design

**Comment by yuval-hazaz** [comment]
Prisma API supports using AND, OR, NOT in the EntityWhereInput 
It should be an easy implementation to add these properties to the DTO

```typeScript
    AND?: Enumerable<PostWhereInput>
    OR?: Enumerable<PostWhereInput>
    NOT?: Enumerable<PostWhereInput>
```

I will add this request to the backlog
